### PR TITLE
refactor(`zetaclient`): use interfaces for used attributes

### PR DESCRIPTION
**Pull request by lumtis**
# Description

Define and configure different interfaces to abstract clients for test purposes to increase the ZetaClient coverage in the future.
Interfaces:
- zetacore bridge
- Bitcoin RPC client
- EVM RPC client

Closes: https://github.com/zeta-chain/node/issues/1154

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Include instructions and any relevant details so others can reproduce. 

- [ ] Tested CCTX in localnet
- [ ] Tested in development environment
- [ ] Go unit tests
- [ ] Go integration tests
- [ ] Tested via GitHub Actions 

# Checklist:

- [ ] I have added unit tests that prove my fix feature works

### CODE REVIEW

**Review by ws4charlie — APPROVED** [review]
LGTM

File: cmd/zetaclientd/start.go [hunk 1]
```diff
@@ -237,7 +237,7 @@ func start(_ *cobra.Command, _ []string) error {
 	}
 
 	// CreateCoreObserver : Core observer wraps the zetacore bridge and adds the client and signer maps to it . This is the high level object used for CCTX interactions
-	mo1 := mc.NewCoreObserver(zetaBridge, signerMap, chainClientMap, metrics, tss, masterLogger, cfg, telemetryServer)
```
Comment (kevinssgh) [inline cmd/zetaclientd/start.go:240]: Is this tss reference not used in the core observer?

**Review by kevinssgh — APPROVED** [review]
Looks good, just left one comment

File: cmd/zetaclientd/start.go (hunk 1 above)
Comment (kevinssgh) [reply cmd/zetaclientd/start.go:240]: I did a quick search and it doesn't seem to be used. You can merge when ready.

**Pull request merged**